src/Microsoft.ML.Data/Commands/TypeInfoCommand.cs

```diff
@@ -0,0 +1,155 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.ML.Data.Commands;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Command;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.Data.Conversion;
+using Microsoft.ML.Runtime.Internal.Utilities;
+
+[assembly: LoadableClass(typeof(TypeInfoCommand), typeof(TypeInfoCommand.Arguments), typeof(SignatureCommand),
+    "", TypeInfoCommand.LoadName)]
+
+namespace Microsoft.ML.Data.Commands
+{
+    public sealed class TypeInfoCommand : ICommand
+    {
+        public const string LoadName = "TypeInfo";
+        public const string Summary = "Displays information about the standard primitive " +
+            "non-key types, and conversions among them.";
+
+        public sealed class Arguments
+        {
+        }
+
+        private readonly IHost _host;
+
+        public TypeInfoCommand(IHostEnvironment env, Arguments args)
+        {
+            Contracts.CheckValue(env, nameof(env));
+            _host = env.Register(LoadName);
+            _host.CheckValue(args, nameof(args));
+        }
+
+        private struct TypeNaInfo
+        {
+            public readonly bool HasNa;
+            public readonly bool DefaultIsNa;
+
+            public TypeNaInfo(bool hasNa, bool defaultIsNa)
+            {
+                HasNa = hasNa;
+                DefaultIsNa = defaultIsNa;
+            }
+        }
+
+        private sealed class KindSetComparer : IEqualityComparer<ISet<DataKind>>
```

Comment:
>KindSet [](start = 29, length = 7)

`SetOfKinds` :) KindSet can be a set that is exceptionally kind, or a set of kinds.